pkg/sql/catalog/descs/factory.go

```diff
@@ -75,7 +90,7 @@ func NewBareBonesCollectionFactory(
 // MakeCollection constructs a Collection for the purposes of embedding.
 func (cf *CollectionFactory) MakeCollection(
```

Comment:
`Make` is used to signify that the collection is not a pointer. Just changing this to return a pointer isn't okay. I think, at this point, I suppose, you ought to just add the `Monitor` argument to `NewCollection` and get rid of this method. At the time it existed to deal with the fact that we were reducing a pointer allocation by inlining the collection in the `connExecutor`. Now that that's out the window, there's no reason for this I don't think.

Reply:
Yes, given you're removing the one primary use case where we had a descs.Collection initialized as a value in a struct, I don't see any reason to retain `MakeCollection`, do you?

Reply:
Yeah I agree. Will remove it.

Reply:
Done